api/decision.go

```diff
@@ -45,8 +45,11 @@ func (h *DecisionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request, next
 		r.URL.Scheme = x.OrDefaultString(r.Header.Get(xForwardedProto),
 			x.IfThenElseString(r.TLS != nil, "https", "http"))
 		r.URL.Host = x.OrDefaultString(r.Header.Get(xForwardedHost), r.Host)
-		r.URL.Path = x.OrDefaultString(strings.SplitN(r.Header.Get(xForwardedUri), "?", 2)[0], r.URL.Path[len(DecisionPath):])
-
+		f := strings.SplitN(r.Header.Get(xForwardedUri), "?", 2)
+		r.URL.Path = x.OrDefaultString(f[0], r.URL.Path[len(DecisionPath):])
+		if len(f) == 2 {
+			r.URL.RawQuery = x.OrDefaultString(f[1], "")
```

Comment:
r.URL.RawQuery = x.OrDefaultString(f[1], r.URL.RawQuery)

This will close all and any loop holes.

Reply:
![image](https://github.com/user-attachments/assets/99718d98-d58f-4cd4-ada1-8b309a3ed455)
![image](https://github.com/user-attachments/assets/2b6e6fda-7200-4c58-840d-5e62c15140a8)
out of index causes panic

Reply:
![image](https://github.com/user-attachments/assets/5cfaed6d-2c7a-4b03-89f5-d57251d03921)
![image](https://github.com/user-attachments/assets/0cb70d88-b8bf-4c83-8601-df91523c5e68)

Reply:
What is the point of your messages?